File: vm/bench/strings.py
def main():
    s = ""
    for i in range(300000):
        s += "item" + str(i) + ","
    parts = s.split(",")
    count = 0
    for p in parts:
        if p.startswith("item1"):
            count += 1
    print(f"{len(s)} {count} {s.index('item299999')} {len(';'.join(parts))}")


main()

File: vm/bench/php/strings.php
<?php

$s = '';
for ($i = 0; $i < 300000; $i++) {
    $s .= 'item'.$i.',';
}
$parts = explode(',', $s);
$count = 0;
foreach ($parts as $p) {
    if (str_starts_with($p, 'item1')) {
        $count++;
    }
}
echo strlen($s).' '.$count.' '.strpos($s, 'item299999').' '.strlen(implode(';', $parts)), "\n";
